### PR TITLE
[chore] Set Up Automated Email Function

### PR DESCRIPTION
**Pull request by plaanty**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Created a new function that will send out an automated email depending on the input parameters.
- This is all done within email.ts

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Everything is within email.ts, so you shouldn't need to look at anything outside of it.
There is an example function call at the bottom of email.ts that you can run from the command line terminal:
`pnpm dlx tsx actions/emails/email.ts`
This should send an email from the senderAddress email to the recipient email, and should actually be received in the sender's inbox.
You can put in your own different parameters to check this by altering the code within the example function call.

If there is something else that you meant by make it "testable", please let me know and I can change the file to match.
Maybe look out for senderEmail and senderAppPassword? Try a few different emails with those if you really want to be certain about it.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

If this is all clean, I can go in and remove the testing call to the function at the bottom of email.ts if necessary.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

https://nodemailer.com/
https://www.youtube.com/watch?v=n8fVC5UFRjA
https://www.youtube.com/watch?v=YFYQucqgvN4

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: package-lock.json
Comment (jinkang-0) [inline package-lock.json:1]: We should not have a `package-lock.json`, it should be a `yaml` file instead. Did you use `npm` by accident?

File: actions/emails/email.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import * as dotenv from 'dotenv';
+import nodemailer from 'nodemailer';
+
+dotenv.config({ path: '.env.local' });
```
[the remark this answers is not in the document]
Comment (plaanty) [reply actions/emails/email.ts:4]: I just know that when I comment out the 1st and 4th lines, and then I try and run a test on the file using the command 
` pnpm dlx tsx actions/emails/email.ts`
then it will return an error like this:
`Error sending email: Error: Missing credentials for "PLAIN"
    at SMTPConnection._formatError (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:809:19)
    at SMTPConnection.login (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:454:38)
    at C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-transport\index.js:272:32
    at SMTPConnection.<anonymous> (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:215:17)
    at Object.onceWrapper (node:events:632:28)
    at SMTPConnection.emit (node:events:518:28)
    at SMTPConnection._actionEHLO (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:1371:14)
    at SMTPConnection._processResponse (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:993:20)
    at SMTPConnection._onData (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:774:14)
    at SMTPConnection._onSocketData (C:\Users\srhud\devProjects\blueprint\adopt-an-inmate\node_modules\.pnpm\nodemailer@7.0.6\node_modules\nodemailer\lib\smtp-connection\index.js:195:44) {
  code: 'EAUTH',
  command: 'API'
}`
But if I put those lines back in, then the message will send. I'm not really sure why though.

File: actions/emails/email.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  process.env.TESTING_EMAIL_USER as string,
+  process.env.TESTING_EMAIL_APP_PASSWORD as string,
+  'NAME',
+  'TEXT',
+  'SUBJECT',
+  process.env.TESTING_EMAIL_RECIPIENT as string,
+).catch(err => {
+  console.error('Error sending email:', err);
+  process.exit(1);
+});
```
[the remark this answers is not in the document]
Comment (plaanty) [reply actions/emails/email.ts:47]: Gotcha, I can fix it without it spouting a type error if I clarify that `senderAddress`, `senderAppPassword`, and `recipient` (the parameters to the autoEmailSender function) are type `string | undefined`, and then manually do a series of checks that ask "if the type of senderAddress or senderAppPassword or recipient are undefined, then throw an error."

File: package-lock.json
Comment (plaanty) [reply package-lock.json:1]: Do you mean pnpm-lock.yaml? I'm not seeing a package-lock.yaml in my file directory

**Review by InayaY3 — COMMENTED** [review]
Just one console.log statement to delete. Also ik you're working on changing the dotenv dependency but otherwise the functionality is all there! yay

**Review by jinkang-0 — APPROVED** [review]
lgtm

File: actions/emails/email.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import nodemailer from 'nodemailer';
+
+export async function autoEmailSender(
+  senderAddress: string | undefined,
```
Comment (ethan-tam33) [inline actions/emails/email.ts:4]: Is there a reason why senderAddress, senderAppPassword, and recipient should have type `string | undefined` instead of just `string`? I imagine that we want these fields always to be defined?

Comment (jinkang-0) [reply actions/emails/email.ts:4]: In my [reply above](https://github.com/calblueprint/adopt-an-inmate/pull/10#discussion_r2407753902), I mentioned a potential solution to throw an error on the base level of the file (outside of the function) if the fields of `process.env` that we need are undefined - that is, in other words, if the specific environment variables required are not set.

There is no way to alter the type of `process.env` or its members, the type is fundamental to the structure of the framework - that environment variables are either strings (if defined), or undefined (if, well, it is not defined).

Comment (ethan-tam33) [reply actions/emails/email.ts:4]: Ah i see makes sense!

**Review by ethan-tam33 — APPROVED** [review]

File: actions/emails/email.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    throw new Error('Sender address is undefined');
+  }
+
+  if (typeof senderAppPassword === 'undefined') {
+    throw new Error('Sender app password is undefined');
+  }
+
+  if (typeof recipient === 'undefined') {
+    throw new Error('Recipient is undefined');
+  }
```
Comment (jinkang-0) [inline actions/emails/email.ts:21]: Flagging this for explicity: instead of allowing undefined values and throwing an error for them, we should just not allow undefined values at all.

Instead, to solve the issue of undefined environment variables, we can just add an if-statement check before calling the function to ensure they are defined, and otherwise throw an error.

**Comment by ethan-tam33** [comment]
Hey Sam, I think you just need to add an if statement like [this](https://github.com/calblueprint/adopt-an-inmate/blob/main/actions/supabase/client.ts#L3). Then we can check for undefined environment variables while still technically having the `string | undefined` typing
